### PR TITLE
Feature: OpenMEV

### PR DESCRIPTION
**Pull request by sambacha**
This closes:
https://github.com/sushiswap/sushiswap-interface/pull/365
https://github.com/sushiswap/sushiswap-interface/pull/374
https://github.com/sushiswap/sushiswap-interface/pull/254
https://github.com/sushiswap/sushiswap-interface/pull/338
https://github.com/sushiswap/sushiswap-interface/pull/375

### Default Behavior:

All transactions submitted using walletconnect enabled wallets will receive MEV protection

### Refunding Behavior

After 2-weeks refunding will be turned on, this will allow 2 types of transaction submissions
- standard MEV protection (no gas refunding)
- arb. based gas refunding (gas refunding)

#### Wallets Supported

```jsonc 
[
  "rainbow",
  "metamask", // separate method of integration OR via web3connect extension  
  "argent",
  "trust",
  "imtoken",
  "pillar",
  "dharma",
  "gnosissafe",
  "ledgerlive",
  "1inch",
  "GridPlus",
  "Torus"
]
```

### Failover

- in the event of RPC timeout, transactions are pushed  to public mempool
- this logic is contained in our load balancing endpoint for sushirelay.com

- a javascript module is available as a generic implementation module (this is not required for the sushi.com namespace, https://github.com/sambacha/web3-rpc-failover)[https://github.com/sambacha/web3-rpc-failover]

### Testing 

[see this repo https://github.com/manifoldfinance/sushi-test-harness](https://github.com/manifoldfinance/sushi-test-harness)

### MetMask

MetaMask integration requires a separate integration approach involving connection via rpc:localhost, we will open another PR for that as it introduces metamask specific logic that might want further testing

Alt: users can add custom rpc information manually 

Signed-off-by: sam bacha <sam@manifoldfinance.com>

### CODE REVIEW

**Comment by sambacha** [comment]
> Curious about the status of this pull request? Any ideas when it will be reviewed and possibly merged?

we are addressing feedback we have gotten

**Comment by cryptosaurusats** [comment]
any reviewers on it? :(

**Comment by sambacha** [comment]
@matthewlilley @omakasebar

**Comment by matthewlilley** [comment]
Reviewing and LGTM

File: src/config/archer.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { ChainId, JSBI } from '@sushiswap/sdk'
+
+import { BigNumber } from '@ethersproject/bignumber'
+
+export const ARCHER_RELAY_URI: { [chainId in ChainId]?: string } = {
+  [ChainId.MAINNET]: 'https://api.archerdao.io/v1/transaction',
```
Comment (cryptosaurusats) [inline src/config/archer.ts:6]: why using ArcherDAO and not  an api from Manifold?

File: locale/de.json [hunk 2]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "Offene Orders",
   "Open Source": "Open Source",
   "Open main menu": "Hauptmenü öffnen",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/de.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/el.json [hunk 3]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "",
   "Open Source": "",
   "Open main menu": "",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/el.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/en.json [hunk 4]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "",
   "Open Source": "",
   "Open main menu": "",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/en.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/es.json [hunk 5]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "Órdenes abiertas",
   "Open Source": "Código abierto",
   "Open main menu": "Abrir menú principal",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/es.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/fa.json [hunk 6]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "سفارش‌های باز",
   "Open Source": "منبع باز",
   "Open main menu": "منوی اصلی را باز کنید",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/fa.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/fr.json [hunk 7]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "Commandes ouvertes",
   "Open Source": "Open source",
   "Open main menu": "Ouvrir le menu principal",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/fr.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/hi.json [hunk 8]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "ऑर्डर्स खोलें",
   "Open Source": "ओपन सोर्स",
   "Open main menu": "मुख्य मेनू खोलें",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/hi.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/it.json [hunk 9]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "Ordini aperti",
   "Open Source": "Open Source",
   "Open main menu": "Apri menu principale",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/it.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/ja.json [hunk 10]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "オープン注文",
   "Open Source": "オープンソース",
   "Open main menu": "メインメニューを開く",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/ja.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/ko.json [hunk 11]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "주문 오픈",
   "Open Source": "오픈 소스",
   "Open main menu": "메인 메뉴 열기",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/ko.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/pt_BR.json [hunk 12]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "Pedidos abertos",
   "Open Source": "Código aberto",
   "Open main menu": "Abra o menu principal",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/pt_BR.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/ro.json [hunk 13]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "Deschidere comenzi",
   "Open Source": "Open Source",
   "Open main menu": "Deschideți meniul principal",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/ro.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/ru.json [hunk 14]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "Открытые заявки",
   "Open Source": "Открытый источник",
   "Open main menu": "Открыть главное меню",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/ru.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/tr.json [hunk 15]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "",
   "Open Source": "",
   "Open main menu": "",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/tr.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/vi.json [hunk 16]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "Lệnh đang mở",
   "Open Source": "Mã nguồn mở",
   "Open main menu": "Mở menu chính",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/vi.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/zh_CN.json [hunk 17]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "开仓订单",
   "Open Source": "开源",
   "Open main menu": "打开主菜单",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/zh_CN.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: locale/zh_TW.json [hunk 18]
```diff
@@ -229,6 +230,7 @@
   "Open Orders": "公開訂單",
   "Open Source": "開源",
   "Open main menu": "打開主菜單",
+  "OpenMEV Eden": "",
```
Comment (cryptosaurusats) [inline locale/zh_TW.json:233]:
```suggestion
  "OpenMEV Fold": "",
```

File: src/components/Settings/index.tsx [hunk 19]
```diff
@@ -103,25 +109,49 @@ export default function SettingsTab({ placeholderSlippage }: { placeholderSlippa
                 toggle={() => (singleHopOnly ? setSingleHopOnly(false) : setSingleHopOnly(true))}
               />
             </div>
-            {/* {chainId == ChainId.MAINNET && (
+            {chainId == ChainId.MAINNET && (
               <div className="flex items-center justify-between">
                 <div className="flex items-center">
                   <Typography variant="sm" className="text-primary">
-                    {i18n._(t`MEV Shield by Archer DAO`)}
+                    {i18n._(t`OpenMEV Eden`)}
```
Comment (cryptosaurusats) [inline src/components/Settings/index.tsx:116]:
```suggestion
                    {i18n._(t`OpenMEV Fold`)}
```

**Comment by sambacha** [comment]
closing this PR in favor of: https://github.com/sushiswap/sushiswap-interface/pull/459 @matthewlilley  @omakasebar

**Comment by sambacha** [comment]
taken over by https://github.com/sushiswap/sushiswap-interface/pull/478

**Pull request closed without merging**